packages/blunderbuss/src/blunderbuss.ts

```diff
@@ -54,158 +56,169 @@ export const sleep = (ms: number) => {
 };
 
 export function blunderbuss(app: Application) {
-  app.on(
-    [
-      'issues.opened',
-      'issues.reopened',
-      'issues.labeled',
-      'pull_request.opened',
-      'pull_request.reopened',
-      'pull_request.labeled',
-    ],
-    async context => {
-      const config = (await context.config(
+  const events = [
+    'issues.opened',
+    'issues.reopened',
+    'issues.labeled',
+    'pull_request.opened',
+    'pull_request.reopened',
+    'pull_request.labeled',
+  ];
+  app.on(events, async context => {
+    let config: Configuration = {};
+    try {
+      // Reading the config requires access to code permissions, which are not
+      // always available for private repositories.
+      config = (await context.config<Configuration>(
         CONFIGURATION_FILE_PATH,
         {}
-      )) as Configuration;
-      const issue = (context.payload.issue
-        ? context.payload.issue
-        : context.payload.pull_request) as Issue;
+      ))!;
+    } catch (err) {
+      err.message = `Error reading configuration: ${err.message}`;
```

Comment:
I believe this will lose the stack trace, I would perhaps instead do `${err.stack}`.

Reply:
Huh - i was modifying the message of the original error specifically so I *would* preserve the call stack!  I think this works right (please let me know if I'm wrong):

```js
try {
  throw new Error('this is busted!');
} catch (e) {
  e.message = `There was a problem: ${e.message}`;
  throw e;
}
```

Leads to:

```
/Users/beckwith/Code/repo-automation-bots/packages/blunderbuss/test.js:5
  throw e;
  ^

Error: There was a problem: this is busted!
    at Object.<anonymous> (/Users/beckwith/Code/repo-automation-bots/packages/blunderbuss/test.js:2:9)
    at Module._compile (internal/modules/cjs/loader.js:1201:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1221:10)
    at Module.load (internal/modules/cjs/loader.js:1050:32)
    at Function.Module._load (internal/modules/cjs/loader.js:938:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47
```

The call stack looks to point to the correct line of code, and I think it just modifies the message part.  Right?

Reply:
This looks good to me, I guess updating the message preserves the stack trace, works for me.